### ISSUE TITLE
Wallet `--data_dir` argument not respected

### ISSUE THREAD

**Issue by casey** [mimblewimble/grin#2529]
`grin wallet` Seems to ignore the `--data_dir DIRECTORY` option. I can cd into a wallet directory and run commands (`wallet info`, for example) but using `--dat_dir` to specify a non-default, non CWD wallet location doesn't work.

To reproduce:

```sh
mkdir new-wallet
cd new-wallet
grin wallet init --here
cd ..
grin wallet --data_dir new-wallet info
grin wallet --data_dir new-wallet/wallet_data info
```

The last two command would fail, when I would expect them to find the newly created wallet.

I'm using v1.0.1 on macOS.

**Comment by hashmap** [comment]
https://github.com/mimblewimble/grin/pull/2517

**Comment by hashmap** [comment]
@casey should be fixed by #2517 feel free to reopen if needed
